network-libp2p/tests/network.rs

```diff
@@ -318,7 +317,7 @@ async fn create_network_with_n_peers(n_peers: usize) -> Vec<Network> {
 
 #[test(tokio::test)]
 async fn connections_stress_and_reconnect() {
-    let peers: usize = 10;
+    let peers: usize = 5;
```

Comment:
Why this change?

Reply:
This test is flaky and with the old number of peers, the network in the CI would progress too slow as there is more going on during network bootstrapping (namely a multi roundtrip process for the Autonat probes per listen adress).

Peers start to disconnect after 30 seconds thus assertions don't hold anymore. With 5 peers it seems the CI is able to keep up.